src/main/java/io/cdap/plugin/batch/source/ftp/FTPFileSystem.java

```diff
@@ -121,15 +129,20 @@ private FTPClient connect() throws IOException {
     client.connect(host, port);
     int reply = client.getReplyCode();
     if (!FTPReply.isPositiveCompletion(reply)) {
-      throw new IOException("Server - " + host
-                              + " refused connection on port - " + port);
+      String errorMessage = String.format("Server - " + host + " refused connection on port - " + port);
+      throw ErrorUtils.getProgramFailureException(
+        new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+        errorMessage, errorMessage, ErrorType.UNKNOWN, true, null);
     } else if (client.login(user, password)) {
       client.setFileTransferMode(FTP.BLOCK_TRANSFER_MODE);
       client.setFileType(FTP.BINARY_FILE_TYPE);
       client.setBufferSize(DEFAULT_BUFFER_SIZE);
     } else {
-      throw new IOException("Login failed on server - " + host + ", port - "
-                              + port);
+      String errorMessage = String.format("Login failed on server - " + host + ", port - " + port);
```

Comment:
Add corrective action in the message `Please verify credentials and ensure FTP server is running.`